tests/contrib/openai_agents/test_openai.py

```diff
@@ -1839,3 +1851,420 @@ async def test_exception_handling(client: Client):
+            instructions="You are a librarian. You should use your tools to source all your information.",
+            tools=[
+                FileSearchTool(
+                    max_num_results=3,
+                    vector_store_ids=["vs_687fd7f5e69c8191a2740f06bc9a159d"],
+                    include_search_results=True,
+                )
+            ],
+        )
+        result = await Runner.run(starting_agent=agent, input=question)
+
+        # A file search was performed
+        assert any(
+            isinstance(item, ToolCallItem)
+            and isinstance(item.raw_item, ResponseFileSearchToolCall)
+            for item in result.new_items
+        )
+        return result.final_output
+
+
+@pytest.mark.parametrize("use_local_model", [True, False])
+async def test_file_search_tool(client: Client, use_local_model):
+    if not use_local_model and not os.environ.get("OPENAI_API_KEY"):
+        pytest.skip("No openai API key")
+
+    new_config = client.config()
+    new_config["plugins"] = [
+        openai_agents.OpenAIAgentsPlugin(
+            model_params=ModelActivityParameters(
+                start_to_close_timeout=timedelta(seconds=30)
+            ),
+            model_provider=TestModelProvider(FileSearchToolModel())
+            if use_local_model
+            else None,
+        )
+    ]
+    client = Client(**new_config)
+
+    async with new_worker(
+        client,
+        FileSearchToolWorkflow,
+    ) as worker:
+        workflow_handle = await client.start_workflow(
+            FileSearchToolWorkflow.run,
+            "Tell me about a side character in the Iliad.",
+            id=f"file-search-tool-{uuid.uuid4()}",
+            task_queue=worker.task_queue,
+            execution_timeout=timedelta(seconds=30),
+        )
+        result = await workflow_handle.result()
+        if use_local_model:
+            assert result == "Patroclus"
+
+
+class ImageGenerationModel(StaticTestModel):
+    responses = [
+        ModelResponse(
+            output=[
+                ResponseFileSearchToolCall(
+                    queries=["side character in the Iliad"],
+                    type="file_search_call",
+                    id="id",
+                    status="completed",
+                    results=[
+                        Result(text="Some scene"),
+                        Result(text="Other scene"),
+                    ],
+                ),
+                ResponseOutputMessage(
+                    id="",
+                    content=[
+                        ResponseOutputText(
+                            text="Patroclus",
+                            annotations=[],
+                            type="output_text",
+                        )
+                    ],
+                    role="assistant",
+                    status="completed",
+                    type="message",
+                ),
+            ],
+            usage=Usage(),
+            response_id=None,
+        ),
+    ]
```

Comment:
I'm a bit new to these tests, but I don't _think_ there's any point in this class, right? And it's  bit confusing because it claims to be something to be doing image generation but the model is instructing the agent to do file searches and responding with the name of a character instead of an image. Better just to do something like

```python
FakeImageGenerationModel = FileSearchToolModel
```

below?